File: CommandSide/Projections/MachineJobProcessingView .js
options({
    resultStreamName: "MachineJobProcessingViewStream"
})
fromAll()
.partitionBy(function(e){
	return e.body.FactoryId + "|" + e.body.MachineId;
})
.when({
    $init: function(){
        return {
            MachineState: null,
            JobId: null,
            JobState: null,
            LastAppliedEventType: null
        }
    },
    MachineStopped: function(s,e){
        s.MachineState = "Stopped";
        s.LastAppliedEventType = e.eventType;
    },
    MachineStarted: function(s,e){
        s.MachineState = "Started";
        s.LastAppliedEventType = e.eventType;
    },
    NewJobStarted: function(s,e){
        s.JobId = e.JobId;
        s.JobState = "Started";
        s.lastAppliedEventType = e.eventType;
    },
    JobCompleted: function(s,e){
        s.JobId = e.JobId;
        s.JobState = "Completed";
        s.lastAppliedEventType = e.eventType;
    }
})
.outputState();

File: CommandSide/SingleEventReactions/EventAddedReactionApp/MachineJobProcessingView.cs
using Shared;

namespace EventAddedReactionApp
{
    internal sealed class MachineJobProcessingView : IView
    {
        public MachineState? MachineState { get; }
        public string JobId { get; }
        public JobState? JobState { get; }
        public string LastAppliedEventType { get; }

        public MachineJobProcessingView(
            MachineState? machineState,
            string jobId,
            JobState? jobState,
            string lastAppliedEventType)
        {
            MachineState = machineState;
            JobId = jobId;
            JobState = jobState;
            LastAppliedEventType = lastAppliedEventType;
        }

        public override string ToString()
        {
            return $"{nameof(MachineState)}: {MachineState}, {nameof(JobId)}: {JobId}, {nameof(JobState)}: {JobState}, {nameof(LastAppliedEventType)}: {LastAppliedEventType}";
        }
    }
    
    internal enum MachineState
    {
        Stopped,
        Started
    }

    internal enum JobState
    {
        Started,
        Completed
    }
}